README.md

```diff
@@ -30,165 +32,555 @@ The two guiding principles are:
+### Code-splitting and `hops`s `Import()` function
+
+You can use code-splitting (or bundle-splitting) to reduce the size of the assets that your users have to download by creating multiple bundles that contain only the code that is actually needed. Under the hood this uses [dynamic imports](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Statements/import#Dynamic_Imports) which webpack will transform into [separate bundles](https://webpack.js.org/guides/code-splitting/). Hops provides support for this with a custom component (called `Import()`) in order to render the actual content on the server-side and render a placeholder during client-side navigation.
+
+**`src/my-component.js`**
 
-By default Hops will find installed presets automatically (only when it is listed as a top-level dependency in your `package.json` - we won't attempt to find presets in dependencies of dependencies).
+```javascript
+import React from 'react';
+export default () => <p>This has been loaded lazily</p>;
+```
 
-If you prefer, you can also explicitly list the presets that you want to use under the `presets` key in your [application configuration](#configuration). This will disable automatic discovery of hops presets.
+**`src/app.js`**
+
+```javascript
+import { render, Import } from 'hops';
+import React from 'react';
+
+const MyLazyComponent = Import('./my-component');
+
+export default render(
+  <div>
+    <h1>Hello World</h1>
+    <MyLazyComponent />
+  </div>
+);
+```
+
+### CLI / npm scripts
+
+The `hops` package acts as a CLI to be used inside [npm scripts](https://docs.npmjs.com/cli/run-script) or [npx](https://medium.com/@maybekatz/introducing-npx-an-npm-package-runner-55f7d4bd282b):
+
+**`package.json`**
+
+```json
+{
+  "scripts": {
+    "start": "hops start",
+    "build": "hops build",
+    "serve": "hops serve"
+  }
+}
+```
+
+#### `hops --help`
+
+The Hops CLI provides short help texts for all its commands. If, for example, you want to see what options are available for the `build` command you can type: `npx hops build --help`.
+
+#### `npm start`
+
+This command will either start the development server or execute a production build and start the production server, depending on the environment variable `NODE_ENV` and/or the `--production` flag.
+
+_The production flag (`--production` or `-p`) is a shortcut for setting the environment variable `NODE_ENV=production`, so these two are interchangable._
+
+- `npm start` starts the development server with hot module reloading, etc.
+- `npm start -- --production` executes a production build and then starts the production server.
+- `npm start -- --production --static` exports static HTML pages for all configured [locations](#default-settings) and then starts the production server (_this only works in tandem with production mode_).
+
+#### `npm run build`
```

Comment:
I think this or the next section ought to mention the `node_modules/.cache/untool/server.js` in some way or the other to make it clear that you need to keep your node_modules around folder when running the application.

This could be interesting for people that want to deploy to platforms like firebase where the node_modules are installed **for you** during deployment. Maybe `serverFile` config should be mentioned somewhere? 

Long story short, we should explain how to run this in production. :D

Reply:
Maybe we could also change the default configuration, so that:
`distDir` and `buildDir` point to `./dist/public` and `serverDir` points to `./dist` or something like that. Then we'll still have the separation between public and private dist files, but everything is inside `./dist`.

Reply:
Jup, iirc @mikebild suggested that as well some time ago and I really like the idea. But not sure if we might miss something important. @dmbch could you think of anything that speaks against doing that?

Reply:
Since we are bundling everything now, this might work ok. I hate the fact that we would be creating a (sparse) dist folder in development, though - and I am not quite sure how those nested folders will work with all kinds of fs related operations (cleaning, syncing etc.).

Reply:
Let us please remove the folder discussion from this PR - I do agree, however, that we ought to mention the need to keep `serverDir` around for deployments.